homeassistant/config_entries.py

```diff
@@ -436,13 +456,29 @@ def async_entries(self, domain: Optional[str] = None) -> List[ConfigEntry]:
                 title=entry['title'],
                 # New in 0.79
                 connection_class=entry.get('connection_class',
-                                           CONN_CLASS_UNKNOWN))
+                                           CONN_CLASS_UNKNOWN),
+                # New in 0.8x
+                options=entry.get('options'))
             for entry in config['entries']]
 
     @callback
-    def async_update_entry(self, entry, *, data):
+    def async_update_entry(
+            self, entry: ConfigEntry, *,
+            data: dict = None, options: dict = None) -> None:
         """Update a config entry."""
-        entry.data = data
+        if not data and not options:
+            return
+
+        if data:
+            entry.data = data
+
+        if options:
+            entry.options = options
```

Comment:
I think that we shouldn't do this, but instead we should create a copy and change that, like this: 

https://github.com/home-assistant/home-assistant/blob/d795410b27896acd5d0d15c6a23262b443774093/homeassistant/helpers/device_registry.py#L185

That way, when an entry is updated and the listener called, we can compare old vs new entry and see what has changed.

Only downside is that I don't know if we can still go down this path, as we maybe already do `entry.data = X` at places?

Reply:
Oh and config entry is not `attr` based, so it's a little trickier.

Reply:
Never mind, thought about it more and also discuss in #20888, let's assign directly.

Reply:
I think we only change data in the upgrade method right now.